packages/web3-eth/src/validation.ts

```diff
@@ -103,3 +119,121 @@ export function isTransactionCall(value: TransactionCall): boolean {
 export function validateTransactionCall(value: TransactionCall) {
 	if (!isTransactionCall(value)) throw new InvalidTransactionCall(value);
 }
+
+export const validateCustomChainInfo = (transaction: Transaction) => {
+	if (transaction.common !== undefined) {
+		if (transaction.common.customChain === undefined) throw new MissingCustomChainError();
+		if (transaction.common.customChain.chainId === undefined)
```

Comment:
according to 1.x doc `networkId` is also not optional , could you verify this as well

Reply:
If not specified by the user, `networkId` was fetch via RPC call in [1.x](https://github.com/ChainSafe/web3.js/blob/8783f4d64e424456bdc53b34ef1142d0a7cee4d7/packages/web3-eth-accounts/src/index.js#L296). However, we don't have an RPC method for `net_version` yet